source/dem/dem.cc

```diff
@@ -969,21 +1214,35 @@ DEMSolver<dim>::solve()
         {
           integrator_object->integrate_half_step_location(
             particle_handler,
-            parameters.lagrangian_physical_properties.g,
+            g,
             simulation_control->get_time_step(),
             torque,
             force,
             MOI);
         }
       else
         {
-          integrator_object->integrate(
-            particle_handler,
-            parameters.lagrangian_physical_properties.g,
-            simulation_control->get_time_step(),
-            torque,
-            force,
-            MOI);
+          if (has_disabled_contacts && contact_build_number > 1)
```

Comment:
Same comment as above

Reply:
It's also a bit weird to always carry the contact_build_number>1. Why not try to agglomerate this check into a small inlined function that is more verbosed.
contacts_are_disabled()
and it would return  has_disabled_contacts && contact_build_number > 1